sorting/merge_sort.c

```diff
@@ -1,17 +1,38 @@
+/**
+ * @file
+ * @brief Implementation of [merge
+ * sort](https://en.wikipedia.org/wiki/Merge_sort) algorithm
+ */
 #include <stdio.h>
 #include <stdlib.h>
 
-void swap(int *a, int *b)  // To swap the variables//
+/**
+ * @addtogroup sorting Sorting algorithms
+ * @{
+ */
+/** Swap two integer variables
+ * @param [in,out] a pointer to first variable
+ * @param [in,out] b pointer to second variable
+ */
+void swap(int *a, int *b)
 {
     int t;
     t = *a;
     *a = *b;
     *b = t;
 }
 
-void merge(int a[], int l, int r, int n)  // To merge //
+/**
+ * @brief Perform merge of segments.
+ *
+ * @param a array to sort
+ * @param l left index for merge
+ * @param r right index for merge
+ * @param n total number of elements in the array
+ */
+void merge(int *a, int l, int r, int n)
 {
-    int *b = (int *)malloc(n * sizeof(int));
+    int *b = (int *)malloc(n * sizeof(int)); /* dynamic memory must be freed */
```

Comment:
/* dynamic memory must be freed */ missing

Reply:
done